File: Java Solution/ProjectEuler5.java
public class ProjectEuler5{

    public static int GCD(int x, int y){

        if (x > y) return GCD(y, x);
        if (y % x == 0) return x;
        return GCD(x, y % x);

    }

    public static int LCM(int x, int y){
        return x * (y / GCD(x, y));
    }

    public static void main(String[] args){

        int multiple = 1;

        for (int i = 2; i <= 20; i++){
            multiple = LCM(i, multiple);
        }

        System.out.println(multiple);

    }

}

File: CPP Solution/ProjectEuler_5.cpp
#include <iostream>
using namespace std;

// Function to calculate the greatest common divisor (GCD) using Euclid's algorithm
int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

// Function to calculate the least common multiple (LCM)
int lcm(int a, int b) {
    return (a / gcd(a, b)) * b; // LCM(a, b) = (a / GCD(a, b)) * b
}


int main(){

    long long ans = 2;
    for (int i = 3; i < 20; i++) {
        ans = lcm(ans, i);
    }

    cout << ans << endl;

    return 0;
}